### DISCUSSION TITLE
Programmatically set css color variables?

### DISCUSSION THREAD

**Discussion by bartimusprimed** [Textualize/textual#1207] [Q&A] [answered]
Not sure if I just missed it in the docs, but is there an easy way to set the $primary or $secondary colors in python?

Right now, I've been checking if the app is dark and if so, then pulling the primary color scheme out of the Default Light/Dark ColorSystem _variables.

Probably something I overlooked; I did notice you can set the variant to be primary or secondary, is this the same as setting the color?

**Comment by davep** [comment, marked as answer]
Currently Textual doesn't have an exposed interface for doing that, but it's something that makes a lot of sense and we do plan on adding; the actual approach not being decided yet though.

**Reply by c-pher** [reply]
Is there any update on this matter? It’s a much-needed feature.

**Reply by davep** [reply]
You probably want to be looking at theme support: https://textual.textualize.io/guide/design/
